source/parser/x86_64/x86_64.cpp

```diff
@@ -20,62 +15,48 @@
 #include "type_checker.hpp"
 
 namespace smeagle::x86_64 {
-
+  namespace detail {
+    struct parameter {
+      std::string name;
+      std::string type_name;
+      std::string location;
+    };
+
+    struct none_t final : public parameter {
```

Comment:
So the idea is that we are storing both representation of facts and abi interface on the level of the parameter? If we add other types of things, we will have to do the same for them?